src/test/java/io/github/w7mike/logic/ProjectsServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package io.github.w7mike.logic;
+
+import io.github.w7mike.JobConfigurationProperties;
+import io.github.w7mike.model.JobGroupsRepository;
+import io.github.w7mike.model.ProjectRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+
+class ProjectsServiceTest {
+
+    @Test
+    @DisplayName("Should throw IllegalStateException when properties allows one groups and other incomplete groups exist")
+    void createGroup_NoMultipleGroupsProperties_And_IncompleteGroupsExists_throwsIllegalStateException() {
+        //given
+        JobGroupsRepository mockGroupRepository = groupRepositoryReturning(true);
+        //and
```

Comment:
I believe that comments  `// and` and `//System under test` are not really needed.
Thanks to simplicity and clear given/when/then separations the test is already very readable.